src/App.js

```diff
@@ -14,11 +15,85 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: [...goodsFromServer],
+    displayList: false,
+    arrayList: [...goodsFromServer],
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
```

Comment:
Let's move all the sortings inside the `GoodsList` component.
It's good to separate the logic. `App` component should only have the button 'start' which will toggle `displayList`.
And according to the `displayList` you should render `GoodsList` component. There you should pass initial goods and sort them inside that component (not inside the `App`)